File: REST API/C-Sharp/EditTicket.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Threading.Tasks;

public class EditTicket
{
    static HttpClient client = new HttpClient(); // One HttpClient instance be reused.
                                                 //HttpClient is used to make WebAPI Request from console application.

    static async Task Main(string[] args)
    {
        string domain = "YOUR_DOMAIN"; //Your bolddesk domain name.

        string apiPath = "/api/v1/tickets/{ticketId}/update_fields?skipDependencyValidation=true"; //Your API path.

        string apiKey = "YOUR_API_KEY"; //Your API key.

        client.BaseAddress = new Uri("https://" + domain); //Generate API path to call.

        client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json")); //Header Value for Content type.

        client.DefaultRequestHeaders.Add("x-api-key", apiKey); //For APIKEY Authentication.

        try
        {
            //Put data to edit ticket.
            var ticketData = new TicketUpdate    //Sample Data's to update ticket fields
            {
                fields = new Dictionary<string, object>()
                {
                    {"priorityId" , 2 },
                    {"resolutionDue", "2021-06-15T04:48:00.000Z" }
                },
                Notes = "Sample Notes"
            };

            HttpResponseMessage response = await client.PutAsJsonAsync(apiPath, ticketData).ConfigureAwait(false);//To send a PUT request as an asynchronous operation to the specified Uri with the given value serialized as JSON.

            var responseData = await response.Content.ReadAsStringAsync().ConfigureAwait(false); //To read the response.

            response.EnsureSuccessStatusCode(); //For Confirms whether the response is success or Not.

            Console.WriteLine(responseData); //Return the response.
        }

        //To catch the error message.
        catch (Exception ex)
        {
            Console.WriteLine("ERROR: ");
            Console.WriteLine("Status Code: " + ex.StatusCode); //Returns the error status.
            Console.WriteLine(ex.Message);//Returns the error message.
            Console.WriteLine(ex.StackTrace);//Returns the stack at the time uncaught exception occurs.  
        }

    }
}

public class TicketUpdate
{
    public Dictionary<string,object> fields { get; set; }
    
    public string Notes { get; set; }
}

File: REST-API-Python/edit_ticket.py
import requests
import traceback

# One Session instance should be reused (similar to a single HttpClient instance).
session = requests.Session()


def main():
    domain = "YOUR_DOMAIN"  # Your bolddesk domain name (e.g., "yourcompany.bolddesk.com")
    api_key = "YOUR_API_KEY"  # Your API key
    ticket_id = "YOUR_TICKET_ID"  # <-- Replace with actual ticket id (e.g., "12345")

    # Your API path (same as C#)
    api_path = f"/api/v1/tickets/{ticket_id}/update_fields"

    # Query string (same as C#)
    params = {
        "skipDependencyValidation": "true"
    }

    base_url = f"https://{domain}".rstrip("/")
    url = f"{base_url}{api_path}"

    # Header Value for Content type + APIKEY Authentication (matches your C# headers)
    session.headers.update({
        "Accept": "application/json",
        "x-api-key": api_key
    })

    try:
        # PUT data to edit ticket fields (matches TicketUpdate with 'fields' dictionary)
        ticket_data = {
            "fields": {
                "priorityId": 2
            }
        }

        # Send PUT request with JSON body (equivalent to PutAsJsonAsync)
        response = session.put(url, params=params, json=ticket_data)

        # Read response text (like ReadAsStringAsync)
        response_text = response.text

        # Ensure success (like EnsureSuccessStatusCode)
        response.raise_for_status()

        # Print response
        print(response_text)

    except requests.exceptions.HTTPError as ex:
        status_code = ex.response.status_code if ex.response is not None else None
        body = ex.response.text if ex.response is not None else ""
        print("ERROR:")
        print("Status Code:", status_code)
        print("Message:", str(ex))
        print("Response Body:", body)
        print("StackTrace:")
        print(traceback.format_exc())

    except Exception as ex:
        print("ERROR:")
        print("Status Code: N/A")
        print("Message:", str(ex))
        print("StackTrace:")
        print(traceback.format_exc())


if __name__ == "__main__":
    main()